### ISSUE TITLE
Mocking cyclic dependencies with `importOriginal` doesn't work in the browser

### ISSUE THREAD

**Issue by sheremet-va** [vitest-dev/vitest#5825]
### Describe the bug

When mocking the cyclic dependency with an `importOriginal`/`vi.importActual` inside, the transformation hangs.

### Reproduction

The same as:

https://github.com/vitest-dev/vitest/blob/839c39f06700144694677f94f8561f4d19a60b51/test/core/test/mocking/cyclic-import-actual.spec.ts#L5

### System Info

```shell
Any
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by kettanaito** [comment]
Am I correct in understanding that the `importOriginal` module request is still mocked when it shouldn't be?

https://github.com/vitest-dev/vitest/blob/839c39f06700144694677f94f8561f4d19a60b51/test/core/test/mocking/cyclic-import-actual.spec.ts#L6

Or is the root cause elsewhere entirely?

**Comment by sheremet-va** [comment]
Yes, it should return the actual module. In `vite-node`, we can track the call stack, but I am not sure yet if it's possible to make it work in the browser. Maybe we can inject some kind of a query to ignore in the mock handler.

**Comment by CaoKha** [comment]
Any updates on this? 
It would be useful to let us see an error message instead of hanging the browser.

**Comment by sheremet-va** [comment]
> Any updates on this? It would be useful to let us see an error message instead of hanging the browser.

Nothing new. Throwing an error would be nice, but I am not sure how to track it yet.

**Comment by todor-a** [comment]
I'm in the process of migrating from jest to vitest and this has caused me so much headache 🤕  Glad I found this issue to at least know what is happening.

Do you think this is worthy of a FYI in the docs?

**Comment by sheremet-va** [comment]
> I'm in the process of migrating from jest to vitest and this has caused me so much headache 🤕 Glad I found this issue to at least know what is happening.
> 
> Do you think this is worthy of a FYI in the docs?

Yes, we can at least add it to the mocking guide. I think there are already some limitations listed.

**Comment by sheremet-va** [comment]
I have found a solution for native ESM in https://github.com/vitest-dev/vitest/pull/9210

It should be possible to apply it in browser mode as well.
